### PR TITLE
Undo binary breaking change

### LINKED ISSUE #16997: Renaming files is a breaking change
In PR #16976, the `SiteServiceCollectionExtensions` class was renamed to `ServiceCollectionExtensions`, which introduced a binary breaking change.

Suppose you are using Package A from NuGet, which targets Orchard Core 2.0 and is compatible with the 2.x series. Meanwhile, your project depends on Orchard Core 2.1-previews and also includes Package A as a dependency.

In this case, you might encounter an exception like the following:

> An unhandled exception was thrown by the application. System.TypeLoadException: Could not load type 'OrchardCore.DisplayManagement.Handlers.SiteServiceCollectionExtensions' from assembly 'OrchardCore.DisplayManagement, Version=2.1.0.0, Culture=neutral, PublicKeyToken=null'.

If Package A were compiled with the 2.1-previews, this issue would not occur. However, since we have no control over external projects (like Package A in this case), renaming public types or files in minor or patch releases is not advisable, as it can introduce breaking changes for users of those dependencies.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #16997

### CODE REVIEW

**Comment by hishamco** [comment]
Where are the binary breaking changes? They are extension methods in the same assembly

**Comment by MikeAlhayek** [comment]
@hishamco check out the updated description.

**Comment by hishamco** [comment]
I'm not sure if it's considered as binary breaking-changes while it's an extension methods, so the class name doesn't matter

File: src/OrchardCore/OrchardCore.DisplayManagement/SiteServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
-public static class ServiceCollectionExtensions
+/// <summary>
+/// Rename this file to ServiceCollectionExtensions in v3.0.
+/// </summary>
+public static class SiteServiceCollectionExtensions
 {
     public static IServiceCollection AddSiteDisplayDriver<TDriver>(this IServiceCollection services)
         where TDriver : class, IDisplayDriver<ISite>
-    => services.AddDisplayDriver<ISite, TDriver>();
+        => services.AddDisplayDriver<ISite, TDriver>();
```
[the remark this answers is not in the document]
Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.DisplayManagement/SiteServiceCollectionExtensions.cs:13]: I have better things to do that picking on spaces. sorry

**Comment by hishamco** [comment]
I'm not sure who this is related to or affects OC. Did you read about binding redirect it might help in your case

**Comment by MikeAlhayek** [comment]
> I'm not sure if it's considered as binary breaking-changes while it's an extension methods, so the class name doesn't matter

We'll it broke me :)


Let say you have Project A on nuget package that use 2.0 assemblies. Also you have project B that uses 2.1-previews and Project A from nuget.

This will give you an exception similar to: 

> An unhandled exception was thrown by the application. System.TypeLoadException: Could not load type 'OrchardCore.DisplayManagement.Handlers.SiteServiceCollectionExtensions' from assembly 'OrchardCore.DisplayManagement, Version=2.1.0.0, Culture=neutral, PublicKeyToken=null'.

If Project A was complied with 2.1 previews, then this wont be a problem.

**Comment by gvkries** [comment]
@hishamco Renaming classes is always breaking, in case of extension methods as well if you invoke them directly.

**Review by sebastienros — APPROVED** [review]

**Comment by sebastienros** [comment]
I vote for only shipping major versions of Orchard!

**Comment by hishamco** [comment]
> Renaming classes is always breaking, in case of extension methods as well if you invoke them directly.

Agree, but I don't think there's someone invoke them directly otherwise this is not make sense in case of extension methods :)

Again it's a breaking change

**Pull request merged**